bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioManager.java

```diff
@@ -44,27 +45,68 @@
      */
     void play(AudioStream audioStream, String sinkId);
 
+    /**
+     * Plays the passed audio stream on the given sink.
+     *
+     * @param audioStream The audio stream to play
+     * @param sinkId The id of the audio sink to use or null
+     * @param volume The volume to be used or null
+     */
+    void play(AudioStream audioStream, String sinkId, PercentType volume);
```

Comment:
As you have mentioned that `volume` and `sinkdId` can be null, what would happen if `audioStream` is null? Shouldn't it throw NPE or IAE? I believe such invariants should be clearly added in the javadoc as it is a public API.

Reply:
That's a good question. Currenlty the `AudioManagerImpl` drops it silently (see https://github.com/eclipse/smarthome/pull/4963/files#diff-93f4fd35ca9478d18d70f62ae3285f19R98). No logging, no exception. I suggest to throw an IAE in this case.

Reply:
Do you think we should add annotations to this API?

Reply:
could be a solution as well